src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,315 @@
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconAlign }) {
+    switch (size) {
+      case 'xsmall':
+        return iconAlign === 'left' ? [0, 0.5, 0, 0] : [0, 0, 0, 0.5];
+      case 'small':
+        return iconAlign === 'left' ? [0, 0.5, 0, 0] : [0, 0, 0, 0.5];
+      case 'medium':
+        return iconAlign === 'left' ? [0, 1, 0, 0] : [0, 0, 0, 1];
+      case 'large':
+        return iconAlign === 'left' ? [0, 1, 0, 0] : [0, 0, 0, 1];
+      default:
+        return iconAlign === 'left' ? [0, 1, 0, 0] : [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  borderRadius({ theme }) {
+    return theme.spacings.xxsmall;
+  },
+};
+
+const StyledButton = styled(TouchableHighlight)`
+  background-color: ${styles.backgroundColor};
+  border-radius: ${styles.borderRadius};
+  border: ${styles.border};
+`;
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconAlign,
+  align,
+  testID,
+}) => {
+  if (typeof children !== 'string' && typeof children !== 'undefined') {
+    throw Error("Only string is allowed as Button's children");
+  }
+
+  const theme = useContext(ThemeContext);
+
+  return (
+    <Flex flexDirection="row" alignItems="center" alignSelf={styles.alignSelf({ align })}>
+      <Space padding={styles.padding({ size, children, theme })}>
+        <StyledButton
+          onPress={onClick}
+          variantColor={variantColor}
+          disabled={disabled}
+          size={size}
+          variant={variant}
+          activeOpacity={1}
+          underlayColor={styles.underlayColor({ variant, variantColor, theme })}
+          align={align}
+          {...automation(testID)}
+        >
+          <React.Fragment>
+            {icon && iconAlign === 'left' && (
+              <Icon
+                name={icon}
+                size={styles.iconSize({ size, children })}
+                fill={styles.fontColor({ variant, variantColor, disabled })}
+              />
+            )}
+            {icon && iconAlign === 'left' && children && (
+              <Space margin={styles.spaceBetween({ size, iconAlign })}>
+                <View />
+              </Space>
+            )}
+            <Text
+              color={styles.fontColor({ variant, variantColor, disabled })}
+              size={styles.fontSize({ size })}
+              align="center"
+              _weight="bold"
+              _letterSpacing={styles.letterSpacing({ size })}
+              _lineHeight={styles.lineHeight({ size })}
+            >
+              {size === 'xsmall' ? children && children.toUpperCase() : children}
```

Comment:
why do you need one extra check for `children`?
`size === 'xsmall' ? children(why this?) && children.toUpperCase() : children`